src/onboarding/onboarding-core-config.ts

```diff
@@ -114,21 +130,26 @@ class OnboardingCoreConfig extends LitElement {
       <div class="row">
         <ha-textfield
           class="flex"
-          .label=${this.hass.localize(
-            "ui.panel.config.core.section.core.core_config.country"
-          )}
+          .label=${
+            this.hass.localize(
+              "ui.panel.config.core.section.core.core_config.country"
+            ) || "Country"
+          }
           name="country"
+          required
           .disabled=${this._working}
           .value=${this._countryValue}
           @change=${this._handleChange}
         ></ha-textfield>
-
         <ha-textfield
           class="flex"
-          .label=${this.hass.localize(
-            "ui.panel.config.core.section.core.core_config.language"
-          )}
+          .label=${
+            this.hass.localize(
+              "ui.panel.config.core.section.core.core_config.language"
+            ) || "Language"
```

Comment:
It is because of a bug/timing issue in `mwc-textfield`, is required is set before a label is set, it will not render the `*` and set the correct classes.